x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/repository/CcrRestoreSourceService.java

```diff
@@ -172,4 +220,66 @@ private void removeSessionForShard(String sessionUUID, IndexShard indexShard) {
             }
         }
     }
+
+    private static class RefCountedCloseable<T extends Closeable> extends AbstractRefCounted {
+
+        private final String name;
+        private final T object;
+
+        private RefCountedCloseable(String name, T object) {
+            super(name);
+            this.name = name;
```

Comment:
why store the name here if it's already accessible in the superclass?

Reply:
I wanted to remove the name from the super class as it is only used to throw an exception. But that is kind of a separate issue that I do not want to do here. So I will adopt your suggestion, use the name from the super class, and now the name is in `AbstractRefCounted` is used for more than just an exception message. Yay.

Reply:
With more changes, this is not relevant as this is not reference counted anymore